models/tags.js

```diff
@@ -0,0 +1,57 @@
+const db = require('../db');
+
+class tags {
+  static async createTag(entry_id,tag) {
+    // First have to find out if the tag already exist and if it does grab the tag id and make an insertion into journal_entries_tag table (the table takes an entry_id and a tag_id). If the tag doesn't exist it will first creat the tag and then do the function mentioned previously. 
```

Comment:
Why did you decide to do this instead of leaving the `UNIQUE` constraint on tags? Doing so would probably simplify this to trying to add the tag without having to check first, if it's there it will just return an error.

Reply:
I took this route because there is an inital check done with SELECT to see if the tag is pre-existiting. If I were to place a unique constraint it wouldn't do anything but plainly be there because I'm effectively checking to make sure a tag is unique by doing the initial SELECT.